### PR TITLE
add numba 0.55 and omniscidb 5.9/5.10 from conda to CI

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title.

* add Numba 0.55 and drop 0.53
* add Omnisci 5.10 and drop 5.6
* add Python 3.10
* Pin ibis-framework to 2.0 to fix a bug with ibis-omniscidb

### CODE REVIEW

**Comment by guilhermeleobas** [comment]
I had to pin ibis-framework to v2.0.0. After a new release of ibis-omniscidb fixing the issue, we can remove the pin.

https://github.com/omnisci/ibis-omniscidb/pull/61

**Comment by guilhermeleobas** [comment]
@pearu, Can I merge this one?

**Review by pearu — APPROVED** [review]
LGTM!

There is one minor nit for the future omniscidb versions (assuming that string comparison does the right thing).

File: .github/workflows/rbc_test.yml [hunk 1]
```diff
@@ -174,11 +173,18 @@ jobs:
...
         if: matrix.os == 'ubuntu-latest' && matrix.omniscidb-from == 'conda'
+        env:
+          OMNISCIDB_VERSION: ${{ matrix.omniscidb-version }}
         run: |
           mkdir data
           mamba run -n omniscidb-env omnisci_initdb data -f
           mamba run -n omniscidb-env omnisci_server --version
-          mamba run -n omniscidb-env omnisci_server --enable-runtime-udf --enable-table-functions \> omniscidb-output.txt 2\>\&1 \& echo \$! \> omniscidb.pid
+          RUN_FLAGS="--enable-runtime-udf --enable-table-functions"
+          if [[ ${OMNISCIDB_VERSION} == "5.10" ]]; then
```
[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply .github/workflows/rbc_test.yml:183]: I don't think this works in bash :)

**Pull request merged**